### ISSUE TITLE
`fvar<double>` `operator/` acting weird

### ISSUE THREAD

**Issue by bbbales2** [stan-dev/math#2307]
## Description

Here is test code:
```cpp
TEST(whatever2, whatever3) {
  using stan::math::fvar;

  fvar<double> theta = 0.25;
  theta.d_ = 1.0;

  // This is correct                                                                                                                                                                                                                                                           
  std::cout << "correct: " << (1.0 / theta).d_ << std::endl;

  // This is wrong                                                                                                                                                                                                                                                             
  size_t N = 1;
  std::cout << "wrong: " << (N / theta).d_ << std::endl;
}
```

Here is the output:
```
correct: -16
wrong: 2.95148e+20
```

#### Current Version:
v3.4.0

**Comment by SteveBronder** [comment]
It's rollover

```cpp
int main() {
    puts("");
    std::cout << -static_cast<size_t>(5)/ 1.5;
    // returns 1.22978e+19
    return 1;
}
```

I think we need to static_cast to a double here or only accept signed types in the signature

https://godbolt.org/z/h9ejKd

**Comment by bbbales2** [comment]
Static cast I think. It'd be weird if it didn't work with unsigned int.

**Comment by bbbales2** [comment]
Fixed #2304
